prow/cmd/deck/main.go

```diff
@@ -33,19 +34,27 @@ import (
 
 	"github.com/NYTimes/gziphandler"
 	"github.com/ghodss/yaml"
+	"github.com/gorilla/sessions"
 	"github.com/sirupsen/logrus"
 
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/github"
 	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/gitoauth"
 	"k8s.io/test-infra/prow/kube"
 	"k8s.io/test-infra/prow/pjutil"
 	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/userdashboard"
 )
 
 var (
-	configPath   = flag.String("config-path", "/etc/config/config", "Path to config.yaml.")
-	buildCluster = flag.String("build-cluster", "", "Path to file containing a YAML-marshalled kube.Cluster object. If empty, uses the local cluster.")
-	tideURL      = flag.String("tide-url", "", "Path to tide. If empty, do not serve tide data.")
-	hookURL      = flag.String("hook-url", "", "Path to hook plugin help endpoint.")
+	configPath       = flag.String("config-path", "/etc/config/config", "Path to config.yaml.")
+	buildCluster     = flag.String("build-cluster", "", "Path to file containing a YAML-marshalled kube.Cluster object. If empty, uses the local cluster.")
+	tideURL          = flag.String("tide-url", "", "Path to tide. If empty, do not serve tide data.")
+	hookURL          = flag.String("hook-url", "", "Path to hook plugin help endpoint.")
+	userUrl          = flag.String("user-url", "", "Path to hook user dashboard endpoint.")
+	gitAppClientFile = flag.String("gitapp-client-file", "/etc/github/app", "Path to the file containing the Git App Client secret.")
```

Comment:
This and `cookie-secret` need to be configurable and default to "".

Reply:
You don't need to unset these but the whole change introduced in this PR needs to be put behind a flag. You should also make sure that you read these files only when the feature flag is true.

Reply:
I guess you can use `--user-url` to gate this feature?

Reply:
nit: --user-url is probably less clear than --oauth-url or something similar